src/core/executor.js

```diff
@@ -1,16 +1,61 @@
+const treekill = require("treekill");
+const {ChildProcess} = require("child_process");
 const spawn = require("cross-spawn");
 const logger = require("../utils/logger");
 const net = require("net");
 const Locale = require("../locales")("service.run");
 const {format} = require("util");
 
-module.exports = {
-  /**
-   *
-   * @param {string} command
-   * @param {object} envs
-   */
-  execute: async function executeCommand(command, envs) {
+const DEFAULT_TIMEOUT = 4000
+class Executor {
+  constructor(options) {
+    const {
+      port,
+      command,
+      envs,
+      silent,
+      timeout,
+    } = options
+
+    this._port = port
+    this._command = command
+    this._envs = envs || {}
+    this._silent =  silent || false
+    this._timeout =  timeout || DEFAULT_TIMEOUT
+    this._isRunning = false
+  }
+
+  get port() {
+    return this._port
+  }
+
+  get command() {
+    return this._command
+  }
+
+  get envs() {
+    return this._envs
+  }
+
+  get silent() {
+    return this._silent
+  }
+
+  get timeout() {
+    return this._timeout
+  }
+
+  get server() {
+    return this._server
+  }
+
+  get isRunning() {
+    return this._isRunning
+  }
```

Comment:
You probably can get rid of this if you don't need it or if no test documents it.

Reply:
Actually they are used within the different methods of the class